c7n/resources/workspaces.py

```diff
@@ -91,3 +93,51 @@ def get_resource_value(self, k, i):
 class KmsFilter(KmsRelatedFilter):
 
     RelatedIdsExpression = 'VolumeEncryptionKey'
+
+
+@resources.register('workspaces-image')
+class WorkspaceImage(QueryResourceManager):
+
+    class resource_type(TypeInfo):
+        service = 'workspaces'
+        enum_spec = ('describe_workspace_images', 'Images', None)
+        arn_type = 'workspaceimage'
+        name = id = 'ImageId'
+        universal_taggable = True
+
+    augment = universal_augment
+
+
+@WorkspaceImage.filter_registry.register('cross-account')
+class WorkspaceImageCrossAccount(CrossAccountAccessFilter):
+
+    schema = type_schema(
+        'cross-account',
+        # white list accounts
+        whitelist_from=ValuesFrom.schema,
+        whitelist={'type': 'array', 'items': {'type': 'string'}})
+
+    permissions = ('workspaces:DescribeWorkspaceImagePermissions',)
+
+    def process(self, resources, event=None):
+        client = local_session(self.manager.session_factory).client('workspaces')
+        allowed_accounts = set(self.get_accounts())
+        results = []
+        for r in resources:
+            found = False
+            try:
+                accts = client.describe_workspace_image_permissions(
+                    ImageId=r['ImageId']).get('ImagePermissions')
+                for a in accts:
+                    account_id = a['SharedAccountId']
+                    if (account_id not in allowed_accounts):
+                        r.setdefault('c7n:CrossAccountViolations', []).append(account_id)
+                        found = True
+                if found:
+                    results.append(r)
+            except Exception as e:
```

Comment:
afaict the only error we should catch here is client.exceptions.ResourceNotFoundException .. ideally the rest just bubble up.